### ISSUE TITLE
[ALBERT] Pretraining mask n gram seems not right

### ISSUE THREAD

**Issue by OleNet** [google-research/google-research#79]
By the note of this line of [code](https://github.com/google-research/google-research/blob/master/albert/create_pretraining_data.py#L469), `the probilities are set to favor longer ngram sequences `, 

but I think after the code of line 469
```python
  ngrams = np.arange(1, FLAGS.ngram + 1, dtype=np.int64)
  pvals = 1. / np.arange(1, FLAGS.ngram + 1)
  pvals /= pvals.sum(keepdims=True)
```

pvals should be ```pvals = array([0.4379562 , 0.2189781 , 0.1459854 , 0.10948905, 0.08759124])``` given ```FLAGS.ngram==5```

and after [line 500](https://github.com/google-research/google-research/blob/master/albert/create_pretraining_data.py#L500),
```
    n = np.random.choice(ngrams[:len(cand_index_set)],
                         p=pvals[:len(cand_index_set)] /
                         pvals[:len(cand_index_set)].sum(keepdims=True))
```

 the code will random choice a n for gram by the pval.
So according to the pval, it will favor shorter ngram sequences, rather than long sequences.

It confused me, so I want to ask is there something I got wrong?

**Comment by dhlee347** [comment]
You can control it with FLAGS.favor_shorter_ngram (line 474-475), but I also think that it was implemented conversely.

>   if FLAGS.favor_shorter_ngram:
    pvals = pvals[::-1]

**Comment by OleNet** [comment]
Thank you for replying.
The work of <SpanBERT: Improving Pre-training by Representing and Predicting Spans> shows that higher probability for shorter gram is useful, so, in general, I think the code of albert seems not wrong. Only the comments and flags confused readers.
